packages/vast-parser/src/VASTParser.ts

```diff
@@ -38,113 +39,117 @@ export default class VASTParser {
-                    duration = durationText.split(':')
+                const durationText = linearNode.getElementsByTagName('Duration').item(0)?.textContent;
+                const duration = durationText ?
+                    durationText.split(':')
                         .map((time, idx) => parseInt(time) * Math.pow(60, 2 - idx))
-                        .reduce((prev, cur) => prev + cur, 0);
-                }
-
-                const mediaFileNode = linearNode?.MediaFiles?.MediaFile;
-                if (!mediaFileNode) return null;
-                const mediaFiles = mapArrayOrElem(mediaFileNode, mediaFile => {
-                    const url = mediaFile._cdata;
-                    const delivery = mediaFile._attributes.delivery;
-                    const type = mediaFile._attributes.type;
-                    const width = parseInt(mediaFile._attributes.width);
-                    const height = parseInt(mediaFile._attributes.height);
-                    const codec = mediaFile._attributes.codec;
-                    const id = mediaFile._attributes.id;
-                    const bitrate = mediaFile._attributes.bitrate ? parseInt(mediaFile._attributes.bitrate) : void 0;
-                    const minBitrate = mediaFile._attributes.minBitrate ? parseInt(mediaFile._attributes.minBitrate) : void 0;
-                    const maxBitrate = mediaFile._attributes.maxBitrate ? parseInt(mediaFile._attributes.maxBitrate) : void 0;
+                        .reduce((prev, cur) => prev + cur, 0)
+                    : 0;
+
+                const mediaFileNodes = Array.from(linearNode.getElementsByTagName('MediaFile'));
+                const mediaFiles = mapArrayOrElem(mediaFileNodes, mediaFile => {
+                    if (mediaFile.parentNode?.nodeName.toLowerCase() !== 'mediafiles') return null;
+
+                    const url = mediaFile.textContent;
+                    const delivery = mediaFile.getAttribute('delivery');
+                    const type = mediaFile.getAttribute('type');
+                    const width = mediaFile.getAttribute('width');
+                    const height = mediaFile.getAttribute('height');
+                    const codec = mediaFile.getAttribute('codec');
+                    const id = mediaFile.getAttribute('id');
+                    const bitrate = mediaFile.getAttribute('bitrate');
+                    const minBitrate = mediaFile.getAttribute('minBitrate');
+                    const maxBitrate = mediaFile.getAttribute('maxBitrate');
                     const newMediaFile = new MediaFile(
-                        url,
-                        delivery,
-                        type,
-                        width,
-                        height,
-                        codec,
-                        id,
-                        bitrate,
-                        minBitrate,
-                        maxBitrate,
+                        url || '',
+                        delivery || '',
```

Comment:
`delivery` from VAST 4.x is required and must be either `"progressive"` or `"streaming"`